src/main/java/org/jenkinsci/plugins/workflow/support/pickles/ExecutorPickle.java

```diff
@@ -53,6 +61,10 @@
 
     private final Queue.Task task;
 
+    private final boolean isEphemeral;
```

Comment:
Definitely you need to track this. If the node was ephemeral, you need not bother waiting at all—it cannot come back, since the very record of its existence has been discarded already.